source/fe/fe_simplex_p.cc

```diff
@@ -100,20 +127,44 @@ namespace
         return unit_points;
       }
 
-    Assert(degree <= 2, ExcNotImplemented());
-    for (const auto &vertex_no : reference_cell.vertex_indices())
-      unit_points.emplace_back(reference_cell.template vertex<dim>(vertex_no));
+    // otherwise write everything as linear combinations of vertices
+    const auto dpo = get_dpo_vector_fe_p(dim, degree);
+    Assert(dpo.size() == dim + 1, ExcInternalError());
+    Assert(dpo[0] == 1, ExcNotImplemented());
+    // vertices:
+    for (unsigned int d : reference_cell.vertex_indices())
+      unit_points.push_back(reference_cell.template vertex<dim>(d));
+    // lines:
+    for (unsigned int l : reference_cell.line_indices())
+      {
+        const Point<dim> p0 =
+          unit_points[reference_cell.line_to_cell_vertices(l, 0)];
+        const Point<dim> p1 =
+          unit_points[reference_cell.line_to_cell_vertices(l, 1)];
+        for (unsigned int p = 0; p < dpo[1]; ++p)
+          unit_points.push_back((double(dpo[1] - p) / (dpo[1] + 1)) * p0 +
+                                (double(p + 1) / (dpo[1] + 1)) * p1);
+      }
+    // quads:
+    if (dim > 1 && dpo[2] > 0)
+      {
+        Assert(dpo[2] == 1, ExcNotImplemented());
+        if (dim == 2)
+          unit_points.push_back(reference_cell.template barycenter<dim>());
+        if (dim == 3)
+          for (unsigned int f : reference_cell.face_indices())
+            unit_points.push_back(face_midpoint<dim>(f));
+      }
 
-    if (degree == 2)
-      for (const auto &line_no : reference_cell.line_indices())
-        {
-          const auto v0 = reference_cell.template vertex<dim>(
-            reference_cell.line_to_cell_vertices(line_no, 0));
-          const auto v1 = reference_cell.template vertex<dim>(
-            reference_cell.line_to_cell_vertices(line_no, 1));
-          unit_points.emplace_back((v0 + v1) / 2.0);
-        }
+    return unit_points;
+  }
 
+  template <>
+  std::vector<Point<0>>
+  unit_support_points_fe_p(const unsigned int /*degree*/)
+  {
+    std::vector<Point<0>> unit_points;
+    unit_points.emplace_back();
     return unit_points;
```

Comment:
I don't understand this code. I believe what you are creating is an array with one element, which is default-initialized. This may be easier to read if you did
```suggestion
    return { Point<0>() };
```